starter/components/locale-switch.js

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState, useRef } from "react"
+import PropTypes from "prop-types"
+import { useRouter } from "next/router"
+
+import WorldIcon from "./icons/world"
+
+import { fetchAPI, getPageData } from "../utils/api"
+import { useOnClickOutside } from "../utils/hooks"
+
+import Cookies from "js-cookie"
+import { MdExpandMore } from "react-icons/md"
+
+const LocaleSwitch = ({ initialLocale }) => {
+  const router = useRouter()
+  const select = useRef()
+  const [locale, setLocale] = useState(initialLocale)
+  const [showing, setShowing] = useState(false)
+  const [pageLocalizations, setPageLocalizations] = useState()
+
+  const handleLocaleChange = async (selectedLocale) => {
+    let slug = router.query
+    if (locale !== router.defaultLocale && !router.query.slug) {
+      // The locale root page is trying to be accessed
+      // The page slug in Strapi should be set to the locale code
+      slug = { slug: [locale] }
+    }
+
+    const currentPage = await getPageData(slug, locale, router.isPreview)
+    const localizations = currentPage ? currentPage.localizations : null
+
+    setPageLocalizations(localizations)
+    // Persist the user's language preference
+    // https://nextjs.org/docs/advanced-features/i18n-routing#leveraging-the-next_locale-cookie
+    Cookies.set("NEXT_LOCALE", selectedLocale)
+    setLocale(selectedLocale)
+  }
+
+  useOnClickOutside(select, () => setShowing(false))
+
+  useEffect(() => {
+    const changeLocale = async () => {
+      setShowing(false)
+
+      if (
+        locale !== initialLocale &&
+        !pageLocalizations?.length &&
+        !router.isPreview
+      ) {
+        // A page was not found for the new locale, redirect to locale root
+        router.push("/", "/", { locale: locale })
+      }
+
+      if (locale !== initialLocale && pageLocalizations?.length) {
+        // Get the localization for the requested locale
+        const localization = pageLocalizations.find(
+          (localization) => localization.locale === locale
+        )
```

Comment:
works in develop, but with a production build the `initialLocale` from the cookie set in `_app` doesn't seem to be updating...

Reply:
This was the problem
```
locale !== initialLocale
```